tests/test_solver.py

```diff
@@ -17,3 +21,39 @@ def test_exp_euler(x_inf):
     fwd_euler = x + dt * vectorfield
     exp_euler = exponential_euler(x, dt, x_inf, x_tau)
     assert np.abs(fwd_euler - exp_euler) / np.abs(fwd_euler) < 1e-4
+
+
+def test_fwd_euler_and_crank_nicolson():
+    """FWD Euler does not yet support branched cells, but comps, branches, nets work.
+
+    Tests whether forward Euler and Crank-Nicolson are sufficiently close to implicit
+    Euler."""
+    comp = jx.Compartment()
+    branch = jx.Branch(comp, nseg=4)
+    branch.insert(HH())
+    cell = jx.Cell(branch, parents=[-1])
+    net = jx.Network([cell for _ in range(2)])
+
+    current = jx.step_current(1.0, 1.0, 0.1, 0.025, 10.0)
+    net.cell(0).branch(0).comp(0).stimulate(current)
+    net.cell(1).branch(0).comp(3).record()
+
+    pre = net.cell(0).branch(0).comp(0)
+    post = net.cell(1).branch(0).comp(0)
+    connect(pre, post, IonotropicSynapse())
+    net.IonotropicSynapse.set("IonotropicSynapse_gS", 0.001)
+
+    # As expected, using significantly shorter compartments or lower r_a leads to NaN
+    # in forward Euler.
+    net.set("length", 50.0)
+    net.set("axial_resistivity", 200.0)
+
+    bwd_v = jx.integrate(net, solver="bwd_euler")
+    fwd_v = jx.integrate(net, solver="fwd_euler")
+    crank_v = jx.integrate(net, solver="crank_nicolson")
+
+    # These allowed voltage differences may appear large, but due to the steepness
+    # of a spike these values actually correspond to sub-millisecond spike time
+    # differences.
+    assert np.max(np.abs(bwd_v - fwd_v)) < 25.0
+    assert np.max(np.abs(bwd_v - crank_v)) < 12.0
```

Comment:
I could see this test pass too easily. Do we have to evaluate on spiking potentials? Might be better to lower the allowed error and evaluate on a simpler trace. If we want to have spikes, I'd suggest adding an additional assert with np.mean or np.std, to make this slightly harder to pass. Just out of caution.

Reply:
I think it's fine, this test is not so easy to pass (e.g. if the AP is missing or off by more than 1ms then the test fails.